src/randoop/operation/Operation.java

```diff
@@ -0,0 +1,121 @@
+package randoop.operation;
+
+import java.io.PrintStream;
+import java.util.List;
+
+import randoop.ExecutionOutcome;
+import randoop.reflection.ReflectionPredicate;
+import randoop.sequence.Variable;
+
+/**
+ * Represents an operation (in the term algebra sense) that can occur in a statement 
+ * as part of a test sequence. Operations include method calls, constructor calls, field
+ * accesses, enum constant values, or primitive values.
+ * 
+ * To support text-based serialization, an implementing class C should also provide:
```

Comment:
Should all implementations do so?  What are the tradeoffs and how does one choose whether to do this?  Or is it required?  In that case, if the method is abstract an implementing class must do so regardless.